Sources/PointFree/Admin/FreeEpisodes.swift

```diff
@@ -0,0 +1,111 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+let indexFreeEpisodeEmailMiddleware =
+  requireAdmin
+    <| writeStatus(.ok)
+    >-> respond(freeEpisodeView.contramap(lower))
+
+private let freeEpisodeView = View<Database.User> { _ in
+  [
+    h2(["Send free episode email"]),
+
+    ul(
+      AppEnvironment.current.episodes()
+        .filter((!) <<< ^\.subscriberOnly)
+        .sorted(by: their(^\.sequence))
+        .map(li <<< freeEpisodeEmailRowView.view)
+    )
+  ]
+}
+
+private let freeEpisodeEmailRowView = View<Episode> { ep in
+  [
+    p([
+      .text(encode(ep.title)),
+      form([action(path(to: .admin(.freeEpisodeEmail(.send(ep.id))))), method(.post)], [
+        input([type(.submit), value("Send email!")])
+        ])
+      ])
+  ]
+}
+
+let sendFreeEpisodeEmailMiddleware:
+  Middleware<StatusLineOpen, ResponseEnded, Tuple2<Database.User?, Episode.Id>, Data> =
+  requireAdmin
+    <<< filterMap(get2 >>> fetchEpisode >>> pure, or: redirect(to: .admin(.freeEpisodeEmail(.index))))
+    <| sendFreeEpisodeEmails
+    >-> redirect(to: .admin(.index))
+
+func fetchEpisode(_ id: Episode.Id) -> Episode? {
+  return AppEnvironment.current.episodes().first(where: { $0.id.unwrap == id.unwrap })
+}
+
+private func sendFreeEpisodeEmails<I>(_ conn: Conn<I, Episode>) -> IO<Conn<I, Prelude.Unit>> {
+
+  return AppEnvironment.current.database.fetchFreeEpisodeUsers()
+    .mapExcept(bimap(const(unit), id))
+    .flatMap { users in
+      sendEmail(forFreeEpisode: conn.data, toUsers: users)
+    }
+    .run
+    .map { _ in conn.map(const(unit)) }
+}
+
+private func sendEmail(forFreeEpisode episode: Episode, toUsers users: [Database.User]) -> EitherIO<Prelude.Unit, Prelude.Unit> {
+
+  // A personalized email to send to each user.
+  let freeEpisodeEmails = users.map { user in
+    lift(IO { inj2(freeEpisodeEmail.view((episode, user))) })
+      .flatMap { nodes in
+        sendEmail(
+          to: [user.email],
+          subject: "Free Point-Free Episode: \(episode.title)",
+          unsubscribeData: (user.id, .newEpisode),
```

Comment:
I'm sending this email under the "new episode" email, but it could maybe also be an announcement email. or a new newsletter. @stephencelis what do you think?

Reply:
I think this is fine for now.